src/main/kotlin/com/jayrave/moshi/pristineModels/Mapper.kt

```diff
@@ -67,19 +67,17 @@ abstract class Mapper<T : Any> {
     private inner class JsonAdapterForMapper : JsonAdapter<T>() {
         override fun fromJson(reader: JsonReader): T {
 
-            // Clear cached values in field mappings
-            fieldMappings.values.forEach { it.clearLastReadValueInCurrentThread() }
+            val mapValues = hashMapOf<String, Any?>()
```

Comment:
The other option I thought of was to store read values on a stack in `FieldMappingImpl` and pop values in a call to `lastReadValueInCurrentThread()`.

Reply:
2. We could also make the `Mapper` _stateless_ thereby avoiding _reference issues_ altogether.  
```kotlin
class PersonMapper: Mapper {
    val name = field(Person::name)
    val age = field(Person::age)
}
```
to
```kotlin
class PersonMapper: Mapper {
    override fun mappers() = mapOf(
         "name" to Person::name,
         "age" to Person::age
    )
}
and make necessary changes in Mapper.kt
```